designs/manager_policy_mapping/policy_mapping.md

```diff
@@ -0,0 +1,95 @@
+# Policy Mapping Workflow with IP/MAC
+
+This document describes the proposed changes for policy mapping workflow with IPs/MACs.
+
+## Background
+
+Currently, we have two ways to determine which policy gets applied to which workload:
+
+- Provide the VNF security manager with encapsulation tag
+- Provide the VNF security manager with IP/MAC
+
+If the manager is supporting [Policy Mapping](https://github.com/opensecuritycontroller/security-mgr-api/blob/master/src/main/java/org/osc/sdk/manager/Constants.java#L70) and [Security Group](https://github.com/opensecuritycontroller/security-mgr-api/blob/master/src/main/java/org/osc/sdk/manager/Constants.java#L60), we provide the manager with the security group name and members (their IPs and MACs) information to bind the policy.
+
+## Constraints and Assumptions
+The appliance security manager will not delete a security group with one or more security group interfaces.
+
+## Design Changes
+`SecurityGroup`: Currently, `SecurityGroup` and `SecurityGroupInterface` have many to many entity mapping. The proposed design will change this relationship to a one to many relationship between the SG and the SGIs
+
+### REST API
+`SecurityGroupInterfaceDto`:
+
+```
+[
+  {
+    "virtualSystemId": 0,
+    "name": "string",
+    "policyId": 0,
+    "managerSecurityGroupId": "string", // This new property is being added to store the security group id returned by the manager
+    "failurePolicyType": "FAIL_OPEN",
+    "order": 0,
+    "policies": [
+      {
+        "id": 0,
+        "parentId": 0,
+        "policyName": "string",
+        "mgrPolicyId": "string",
+        "mgrDomainId": 0,
+        "mgrDomainName": "string"
+      }
+    ],
+    "markedForDeletion": false,
+    "binded": false
+  }
+]
+```
+
+### OSC SDKs
+
+#### VNF Security Manager SDK
+
+`SecurityGroupInterfaceElement`: Add a new interface
+
+```
+public interface SecurityGroupInterfaceElement {
+
+    /**
+	 * Name of the security group
+	 */
+    String getName();
+
+	/**
+	 * Provide manager securty group if the manager is supporting policy mapping and security group
+	 */
+    String getManagerSecurityGroupId();
+
+	/**
+	 * Provide tag in case manager is not supporting security group
```

Comment:
This is incorrect
tag is the manager policy id in case of IP/MAC
its the encapsulation tag in case of encapsulation